programs/mango-v4/src/state/mango_account.rs

```diff
@@ -899,6 +906,10 @@ impl<
         let (base_change, quote_change) = fill.base_quote_change(side);
         let quote = cm!(I80F48::from(perp_market.quote_lot_size) * I80F48::from(quote_change));
         let fees = cm!(quote.abs() * I80F48::from_num(fill.maker_fee));
+        if fees.is_positive() {
+            self.fixed_mut().discount_buyback_fees_accrued += fees.round().to_num::<u64>();
+        }
+        let pa = self.perp_position_mut(perp_market_index)?;
```

Comment:
looks like `pa` is unused above, move the whole thing (including settle_funding) here?